src/UserSpaceInstrumentation/AccessTraceesMemory.cpp

```diff
@@ -25,44 +26,61 @@ using orbit_base::ReadFileToString;
 [[nodiscard]] ErrorMessageOr<std::vector<uint8_t>> ReadTraceesMemory(pid_t pid,
                                                                      uint64_t address_start,
                                                                      uint64_t length) {
-  // Round up length to next multiple of eight.
-  length = ((length + 7) / 8) * 8;
-  std::vector<uint8_t> bytes(length);
-  for (size_t i = 0; i < length / 8; i++) {
-    const uint64_t data = ptrace(PTRACE_PEEKDATA, pid, address_start + i * 8, NULL);
-    if (errno) {
-      return ErrorMessage(
-          absl::StrFormat("Failed to PTRACE_PEEKDATA for pid %d with errno %d: \"%s\"", pid, errno,
-                          SafeStrerror(errno)));
-    }
-    std::memcpy(bytes.data() + (i * sizeof(uint64_t)), &data, 8);
+  if (length == 0) {
+    return ErrorMessage("Tried to read 0 bytes.");
   }
-  return bytes;
+
+  OUTCOME_TRY(fd, orbit_base::OpenFileForReading(absl::StrFormat("/proc/%d/mem", pid)));
+
+  std::vector<uint8_t> data(length);
+  ssize_t result = pread(fd.get(), data.data(), length, address_start);
```

Comment:
maybe implement orbit_base::ReadFullyAtOffset() note that pread can return value less then length in case it was interrupted by signal, so we want to just continue reading in this case.